tests/models/test_neuralgcm.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import pytest
+
+pytest.importorskip('neuralgcm', minversion='0.2.0')
+pytest.importorskip('jax')
+pytest.importorskip('xarray')
+pytest.importorskip('gcsfs')
+
+from torchgeo.models import NeuralGCM_Weights, neuralgcm
+
+
+class TestNeuralGCM:
+    def test_invalid(self) -> None:
+        with pytest.raises(ValueError, match='Please specify a weights enum'):
+            neuralgcm()
+
+    @pytest.mark.slow
+    @pytest.mark.parametrize(
+        'weights',
+        [
+            NeuralGCM_Weights.DETERMINISTIC_0_7_DEG,
+            NeuralGCM_Weights.DETERMINISTIC_1_4_DEG,
+            NeuralGCM_Weights.DETERMINISTIC_2_8_DEG,
+            NeuralGCM_Weights.STOCHASTIC_1_4_DEG,
+            NeuralGCM_Weights.STOCHASTIC_PRECIP_2_8_DEG,
+            NeuralGCM_Weights.STOCHASTIC_EVAP_2_8_DEG,
+        ],
+    )
+    def test_weights(self, weights: NeuralGCM_Weights) -> None:
+        model = neuralgcm(weights=weights)
+        assert hasattr(model, 'timestep')
+        assert hasattr(model, 'data_coords')
+        assert hasattr(model, 'model_coords')
+        assert hasattr(model, 'input_variables')
+        assert hasattr(model, 'forcing_variables')
+
+    @pytest.mark.slow
+    def test_default_weights(self) -> None:
+        model = neuralgcm(weights=NeuralGCM_Weights.DETERMINISTIC_1_4_DEG)
+        assert hasattr(model, 'timestep')
+        assert hasattr(model, 'encode')
+        assert hasattr(model, 'decode')
+        assert hasattr(model, 'advance')
+        assert hasattr(model, 'unroll')
+
+    @pytest.mark.slow
+    def test_forecasting_integration(self) -> None:
+        """Test NeuralGCM forecasting workflow using ERA5 data from GCS.
+
+        Performs parts of the NeuralGCM quick start guide:
+        https://neuralgcm.readthedocs.io/en/latest/inference_demo.html
+        """
+        import jax
+        import numpy as np
+        import xarray as xr
+        from dinosaur import horizontal_interpolation, spherical_harmonic, xarray_utils
+
+        # Load model
+        model = neuralgcm(weights=NeuralGCM_Weights.DETERMINISTIC_2_8_DEG)
+
+        # Load ERA5 data
+        era5_path = (
```

Comment:
Our tests are all offline so we don't want to download sample data. Can you instead make a dummy xarray dataset with the same variables?